Clase-09/fcfg.ipynb

```diff
@@ -0,0 +1,1418 @@
+   "metadata": {},
+   "source": [
+    "Sin embargo, esto no es una función. Veamos cómo sería implementar una función."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fstruct(attribute):\n",
+    "    value = fs_b_nltk[attribute]\n",
+    "    return value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fstruct('CONC')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Esta función, no obstante, no nos permite \"caminar\" por la estructura. Solo toma un único atributo y devuelve su valor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fstruct('CONC','GEN')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Usando un asterisco (\\*) antes del parámetro `attribute` podemos indicarle a la función que reciba más de un atributo. Todos los atributos que le indiquemos, serán guardados en una tupla sobre la que podremos iterar."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def example(*args):\n",
+    "    return args"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "example('a','b','c')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fstruct(*attributes):\n",
+    "    value = fs_b_nltk\n",
+    "    for attr in attributes:\n",
+    "        value = value[attr]\n",
+    "    return value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fstruct('CONC')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fstruct('CONC','GEN')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "En las funciones anteriores, la estructura de rasgos se encontraba _hardcodeada_. Es decir que estaba especificada dentro de la función y el usuario no tiene posibilidad de modificarla si, por ejemplo, quisiera conocer el valor de un atributo en otra estructura. Para evitar esto, es posible convertirla en un parámetro más."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fstruct(fs, *attributes):\n",
+    "    value = fs\n",
+    "    for attr in attributes:\n",
+    "        value = value[attr]\n",
+    "    return value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fstruct(fs_b_nltk, 'CONC', 'NUM')"
```

Comment:
esta celda me da error. 
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Input In [58], in <cell line: 1>()
----> 1 fstruct(fs_c_nltk, 'CONC', 'NUM')

Input In [56], in fstruct(fs, *attributes)
      2 value = fs
      3 for attr in attributes:
----> 4     value = value[attr]
      5 return value

File ~/.local/lib/python3.8/site-packages/nltk/featstruct.py:646, in FeatDict.__getitem__(self, name_or_path)
    643 """If the feature with the given name or path exists, return
    644 its value; otherwise, raise ``KeyError``."""
    645 if isinstance(name_or_path, (str, Feature)):
--> 646     return dict.__getitem__(self, name_or_path)
    647 elif isinstance(name_or_path, tuple):
    648     try:

KeyError: 'CONC'

Reply:
M... No sé por qué... A mí me anda. Y el .md también lo corrió ok (el script que genera el .md corre todas las celdas de nuevo, para que queden numeradas bien).